editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5925,12 +5927,24 @@ bool CanvasItemEditorViewport::can_drop_data(const Point2 &p_point, const Varian
 				if (!instantiated_scene) {
 					continue;
 				}
+
+				Node *edited_scene = EditorNode::get_singleton()->get_edited_scene();
+				if (_cyclical_dependency_exists(edited_scene->get_scene_file_path(), instantiated_scene)) {
+					memdelete(instantiated_scene);
+					can_instantiate = false;
+					is_cyclical_dep = true;
+					break;
+				}
 				memdelete(instantiated_scene);
 			}
 			can_instantiate = true;
-			break;
 		}
 	}
+	if (is_cyclical_dep) {
+		canvas_item_editor->message = TTR("Can't instantiate: ") + TTR("Cyclical dependency found");
```

Comment:
```suggestion
		canvas_item_editor->message = vformat(TTR("Can't instantiate: %s."), TTR("Cyclical dependency found"));
```
This message is a nice touch. Maybe we could have one for other problems too.

Reply:
I'd say the correct form of the word is "cyclic", "cyclical" specifically refers to time periods

Reply:
Yes, I used "cyclical" as that's how the method was called. I've seen "Circular dependency" in some issues here. I could also attach the filename(s) of the circular dependent files? what do you think?

Reply:
The established wording in the engine is "circular", see:
https://github.com/godotengine/godot/blob/9b94c80e9aff2a4f363ae6d8e2bbe837aa5876bc/core/io/resource_format_binary.cpp#L1973

https://github.com/godotengine/godot/blob/9b94c80e9aff2a4f363ae6d8e2bbe837aa5876bc/scene/resources/resource_format_text.cpp#L1874

https://github.com/godotengine/godot/blob/9b94c80e9aff2a4f363ae6d8e2bbe837aa5876bc/tests/core/io/test_resource.h#L143

Reply:
Sure, will change.
What do you think about including the file(s) that contain circular dependencies? downsides:

- If there is a lot, the message could get really long.
- The loop will always iterate over all files.

Reply:
I'd say add the file that fails, not the files that break it, so if "foo" contains circular references mention "foo"

Reply:
Yes, that's what I meant. But I was talking about the possibility that multiple files contain circular deps; though giving it a bit more thought it feels more over-complicated to implement

Reply:
Does this look good?

![image](https://github.com/godotengine/godot/assets/55825613/5f3492c4-1413-4dd5-a552-4de68218c704)